### PR TITLE
Added Seed in ModelParams for Summarization Requests

### PR DESCRIPTION
**Pull request by faiqibrahim**
# Heads up!
These protobufs in this repository are automatically generated from Vectara and
the repository is read-only for the general public.  As a result, we do not
accept pull requests to this GitHub repository.

If you have feedback on Vectara or its APIs, please let us know in our
forums: https://discuss.vectara.com/

### CODE REVIEW

File: serving.proto [hunk 1]
```diff
@@ -112,9 +122,15 @@ message SummarizationRequest {
...
     optional float presence_penalty = 20;
+    // This feature is in Beta. If specified, the LLM will make a best effort to sample deterministically,
+    // such that repeated requests with the same seed and parameters should return the same result.
+    // Note:- Seed value is only used when temperature is set to 0.
+    optional uint32 seed = 25;
   }
   ModelParams model_params = 215;
 
+  // Parameters for the citation style.
+  CitationParams citation_params = 220;
```
Comment (tallatshafaat) [inline serving.proto:133]: Not all prompts support citation prompts. @pwoznic we should have a list somewhere in the docs that lists which prompts support this.

File: admin_apikey.proto [hunk 2]
```diff
@@ -47,6 +47,8 @@ message ApiKey {
   optional int64 ts_end = 6;
   // Status of the ApiKey.
   ApiKeyStatus status = 7;
+  // An id of the key that is not sensitive.
+  uint32 public_id = 8;
```
Comment (tallatshafaat) [inline admin_apikey.proto:51]: @nikron are you okay making this part of the public api? i don't see a reason why not, but just FYI, this is going public.

**Comment by tallatshafaat** [comment]
I don't know what's up with permission changes to these proto files.

**Review by tallatshafaat — COMMENTED** [review]
I don't know what's up with these proto file permission changes.

From my side, this is good. @pwoznic whenever you're ready to make this public, please approve and merge. Thanks.

File: serving.proto (hunk 1 above)
Comment (pwoznic) [reply serving.proto:133]: The [citations content](https://docs.vectara.com/docs/api-reference/search-apis/search#citation-format-in-summary) lists the supposed prompts (5 at the moment).

**Review by pwoznic — APPROVED** [review]

**Pull request merged**